scipy/integrate/_quadrature.py

```diff
@@ -1035,3 +1039,227 @@ def newton_cotes(rn, equal=0):
+    increasing `n_estimates` tends to decrease the error estimate.
+
+    Examples
+    --------
+    QMC quadrature is particularly useful for computing integrals in higher
+    dimensions. An example integrand is the probability density function
+    of a multivariate normal distribution.
+
+    >>> import numpy as np
+    >>> from scipy import stats
+    >>> dim = 8
+    >>> mean = np.zeros(dim)
+    >>> cov = np.eye(dim)
+    >>> def func(x):
+    ...     return stats.multivariate_normal.pdf(x, mean, cov)
+
+    To compute the integral over the unit hypercube:
+
+    >>> from scipy.integrate import qmc_quad
+    >>> a = np.zeros(dim)
+    >>> b = np.ones(dim)
+    >>> rng = np.random.default_rng()
+    >>> qrng = stats.qmc.Halton(d=dim, seed=rng)
+    >>> n_estimates = 8
+    >>> res = qmc_quad(func, a, b, n_estimates=n_estimates, qrng=qrng)
+    >>> res.integral, res.standard_error
+    (0.00018441088533413305, 1.1255608140911588e-07)
+
+    A two-sided, 99% confidence interval for the integral may be estimated
+    as:
+
+    >>> t = stats.t(df=n_estimates-1, loc=res.integral,
+    ...             scale=res.standard_error)
+    >>> t.interval(0.99)
+    (0.00018401699720722663, 0.00018480477346103947)
+
+    Indeed, the value reported by `scipy.stats.multivariate_normal` is
+    within this range.
+
+    >>> stats.multivariate_normal.cdf(b, mean, cov, lower_limit=a)
+    0.00018430867675187443
+
+    """
+    args = _qmc_quad_iv(func, a, b, n_points, n_estimates, qrng, log)
+    func, a, b, n_points, n_estimates, qrng, rng, log, stats = args
+
+    # The sign of the integral depends on the order of the limits. Fix this by
+    # ensuring that lower bounds are indeed lower and setting sign of resulting
+    # integral manually
+    if np.any(a == b):
+        message = ("A lower limit was equal to an upper limit, so the value "
+                   "of the integral is zero by definition.")
+        warnings.warn(message, stacklevel=2)
+        return QMCQuadResult(-np.inf if log else 0, 0)
+
+    i_swap = b < a
+    sign = (-1)**(i_swap.sum(axis=-1))  # odd # of swaps -> negative
+    a[i_swap], b[i_swap] = b[i_swap], a[i_swap]
+
+    A = np.prod(b - a)
+    dA = A / n_points
+
+    estimates = np.zeros(n_estimates)
+    rngs = _rng_spawn(qrng.rng, n_estimates)
+    for i in range(n_estimates):
+        # Generate integral estimate
+        sample = qrng.random(n_points)
+        x = stats.qmc.scale(sample, a, b)
+        integrands = func(x)
+        if log:
+            estimate = logsumexp(integrands) + np.log(dA)
+        else:
+            estimate = np.sum(integrands * dA)
+        estimates[i] = estimate
+
+        # Get a new, independently-scrambled QRNG for next time
+        qrng = type(qrng)(seed=rngs[i], **qrng._init_quad)
+
+    integral = np.mean(estimates)
+    integral = integral + np.pi*1j if (log and sign < 0) else integral*sign
+    standard_error = stats.sem(estimates)
```

Comment:
This evaluation could also be vectorized (most useful if here is a high "fixed cost" of evaluating the function):
```suggestion
    rngs = _rng_spawn(qrng.rng, n_estimates)
    samples = []
    for i in range(n_estimates):
        # Generate samples for integral estimate
        samples.append(qrng.random(n_points))
        # Get a new, independently-scrambled QRNG for the next samples
        qrng = type(qrng)(seed=rngs[i], **qrng._init_quad)
    # Evaluate the function in a vectorized fashion
    samples = np.concatenate(samples)
    x = stats.qmc.scale(samples, a, b)
    integrands = func(x).reshape(n_estimates, -1)
    # Compute the estimates (also in a vectorized fashion)
    if log:
        estimates = logsumexp(integrands, axis=-1) + np.log(dA)
    else:
        estimates = np.sum(integrands * dA, axis=-1)

    integral = np.mean(estimates, axis=0)
    integral = integral + np.pi*1j if (log and sign < 0) else integral*sign
    standard_error = stats.sem(estimates, axis=0)
```

Reply:
`n_estimates` is always pretty small (order of 10 or so, just enough to get a reasonable `standard_error` estimate). The overhead of the extra function calls competes with the overhead of the memory copying in the stacked case. The current `for` loop is also a place where we are looking to introduce parallelism later down the road. So I'd keep the `for` loop.

Reply:
Fair enough, I guess it really depends on the particular runtime/memory behavior of `func`. I guess if `func` has a high fixed cost overhead that independent of `x`, then it may be possible to split `func` by pre-computing the fixed cost parameter `theta_x` and then having a function `x -> func_light(x; theta_x)`. Though that may not always be possible, e.g. if `func` is evaluated through some propriety external code with high startup cost.